searx/engines/piratebay.py

```diff
@@ -47,50 +53,43 @@ def request(query, params):
 def response(resp):
     results = []
 
-    dom = html.fromstring(resp.text)
-
-    search_res = dom.xpath('//table[@id="searchResult"]//tr')
+    search_res = loads(resp.text)
 
     # return empty array if nothing is found
-    if not search_res:
+    if search_res[0]["name"] == "No results returned":
         return []
 
     # parse results
-    for result in search_res[1:]:
-        link = result.xpath('.//div[@class="detName"]//a')[0]
-        href = urljoin(url, link.attrib.get('href'))
-        title = extract_text(link)
-        content = extract_text(result.xpath(content_xpath))
-        seed, leech = result.xpath('.//td[@align="right"]/text()')[:2]
-
-        # convert seed to int if possible
-        if seed.isdigit():
-            seed = int(seed)
-        else:
-            seed = 0
-
-        # convert leech to int if possible
-        if leech.isdigit():
-            leech = int(leech)
-        else:
-            leech = 0
-
-        magnetlink = result.xpath(magnet_xpath)[0]
-        torrentfile_links = result.xpath(torrent_xpath)
-        if torrentfile_links:
-            torrentfile_link = torrentfile_links[0].attrib.get('href')
-        else:
-            torrentfile_link = None
+    for result in search_res:
+        link = url + "description.php?id=" + result["id"]
+        magnetlink = "magnet:?xt=urn:btih:" + result["info_hash"] + "&dn=" + result["name"]
```

Comment:
Would it better with a ```\``` at the end of the line ?

Reply:
@dalf That's why I opened a PR in order to fix that: https://github.com/asciimoo/searx/pull/2134 and I'm waiting for someone to merge it :)